libraries/chain/include/eosio/chain/block_state.hpp

```diff
@@ -29,6 +29,7 @@ namespace eosio::chain {
       bool                   is_valid()          const { return validated; } 
       void                   set_valid(bool b)         { validated = b; }
       uint32_t               irreversible_blocknum() const { return 0; } // [greg todo] equivalent of dpos_irreversible_blocknum
+      std::optional<qc_data_t> get_best_qc()     const; 
```

Comment:
We should return `std::optional<quorum_certificate>`.

I don't see the purpose of `qc_data_t` at all. It should be possible to derive a `qc_info_t` from a `quorum_certificate`.